src/AccessibilityInsights.SetupLibrary/GitHubWrapper.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using AccessibilityInsights.SetupLibrary.REST;
+using System;
+using System.Globalization;
+using System.IO;
+
+namespace AccessibilityInsights.SetupLibrary
+{
+    /// <summary>
+    /// Concrete implementation of IGitHubWrapper
+    /// </summary>
+    public class GitHubWrapper : IGitHubWrapper
+    {
+        private readonly Uri _productionConfigFileUri;
+        private readonly Uri _insiderConfigFileUri;
+        private readonly Uri _canaryConfigFileUri;
+        private readonly TimeSpan _timeout;
+
+        private const string DefaultProductionConfigFileUrl = "https://www.github.com/Microsoft/accessibility-insights-windows/blob/Control/Channels/Production/release_info.json?raw=true";
+        private const string DefaultInsiderConfigFileUrl = "https://www.github.com/Microsoft/accessibility-insights-windows/blob/Control/Channels/Insider/release_info.json?raw=true";
+        private const string DefaultCanaryConfigFileUrl = "https://www.github.com/Microsoft/accessibility-insights-windows/blob/Canary/Channels/Canary/release_info.json?raw=true";
+
+        /// <summary>
+        /// ctor
+        /// </summary>
+        /// <param name="exceptionReporter">Mechanism for reporting recoverable exceptions from the config</param>
+        public GitHubWrapper(IExceptionReporter exceptionReporter)
+        {
+            const double defaultTimeout = 60.0;
+            OverridableConfig config = new OverridableConfig("GitHubWrapper.settings", exceptionReporter);
+
+            _productionConfigFileUri = GetConfiguredUri(config, "ProductionConfigFileUrl", DefaultProductionConfigFileUrl);
+            _insiderConfigFileUri = GetConfiguredUri(config, "InsiderConfigFileUrl", DefaultInsiderConfigFileUrl);
+            _canaryConfigFileUri = GetConfiguredUri(config, "CanaryConfigFileUrl", DefaultCanaryConfigFileUrl);
+
+            string configTimeout = config.GetConfigSetting("TimeoutInSeconds", string.Empty);
+            if (!double.TryParse(configTimeout, NumberStyles.Number, CultureInfo.InvariantCulture, out double timeoutInSeconds) || timeoutInSeconds <= 0)
+                timeoutInSeconds = defaultTimeout;
+
+            _timeout = TimeSpan.FromSeconds(timeoutInSeconds);
+        }
+
+        /// <summary>
+        /// Implements <see cref="IGitHubWrapper.LoadUriContentsIntoStream(Uri, Stream)"/>
+        /// </summary>
+        public void LoadUriContentsIntoStream(Uri uri, Stream stream)
```

Comment:
is this method necessary? I'm confused how it relates to the one after it (`LoadChannelInfoIntoStream`) and whether they can be collapsed